packages/start/src/userAgent.ts

```diff
@@ -0,0 +1,13 @@
+import { useRequest } from "solid-start/server";
+import { isServer } from "solid-js/web";
+/** A primitive that allows for the user agent string to be accessed isomorphically on the client, or on the server
+ * @return Returns the user agent string, or null
+ */
+export function useUserAgent() {
+  const event = useRequest();
```

Comment:
This only works in Solid-Start. Please provide a way for users to add their own function that provides the request headers and keep useRequest as a default.

Reply:
I think that was the point
To have a package with hard start dependency

Reply:
OK, everyone else can use the storage primitive anyways.